modules/edu.gemini.seqexec.web/edu.gemini.seqexec.web.server/src/main/scala/edu/gemini/seqexec/web/server/http4s/JwtAuthentication.scala

```diff
@@ -1,23 +1,63 @@
 package edu.gemini.seqexec.web.server.http4s
 
+import java.time.Instant
+
 import edu.gemini.seqexec.model.UserDetails
 import edu.gemini.seqexec.web.server.security.AuthenticationService
-import org.http4s.AttributeKey
+import org.http4s.{AttributeKey, Cookie, HttpService, Response}
 
 import scalaz._
 import Scalaz._
 import scalaz.concurrent.Task
 
+
+trait CookiesService {
+  def name: String
+  def ssl: Boolean
+  def ttl: Long
+
+  def buildCookie(token: String): Cookie = {
+    // if successful set a cookie
+    val exp = Instant.now().plusSeconds(ttl)
```

Comment:
That's a valid concern. I always forget that time is IO. Should be fairly straightforward to wrap it on a Task